modules/aggregation-layers/src/utils/bin-sorter.js

```diff
@@ -23,37 +23,42 @@
 // this is where to pass in a function to color the bins by
 // avg/mean/max of specific value of the point
 const defaultGetValue = points => points.length;
+const MAX_32_BIT_FLOAT = 3.402823466e38;
 
 export default class BinSorter {
-  constructor(bins = [], getValue = defaultGetValue) {
-    this.sortedBins = this.getSortedBins(bins, getValue);
-    this.maxCount = this.getMaxCount();
+  constructor(bins = [], getValue = defaultGetValue, sort = true) {
+    this.aggregatedBins = this.getAggregatedBins(bins, getValue, sort);
+    this._updateMinMaxValues();
     this.binMap = this.getBinMap();
   }
 
   /**
-   * Get an array of object with sorted values and index of bins
+   * Get an array of object with aggregated values and index of bins
+   * Array object will be sorted by value optionally.
    * @param {Array} bins
    * @param {Function} getValue
+   * @param {Bool} sort
    * @return {Array} array of values and index lookup
    */
-  getSortedBins(bins, getValue) {
-    return bins
-      .reduce((accu, h, i) => {
-        const value = getValue(h.points);
+  getAggregatedBins(bins, getValue, sort) {
+    const aggregatedBins = bins.reduce((accu, h, i) => {
```

Comment:
- Is this a public method? Can the user update an existing instance?
- Use for loop